src/tests/ftest/soak/soak.py

```diff
@@ -20,287 +20,433 @@
+
+    def remote_copy(self, hostlist, remote_dir, local_dir):
+        """Copy files from remote dir to local dir.
+
+        This is a temporary method and will be replaced by
+        clush in general_utils
+        Args:
+                hostlist (list): list of remote nodes
+                remote_dir (str): remote directory of files
+                local_dir (str): local directory
+
+        Returns:
+            status: bool
 
-        self.pool = DaosPool(self.context)
-        self.pool.create(createmode, createuid, creategid,
-                         createsize, createsetid, None, None,
-                         self.createsvc)
+        """
+        this_host = socket.gethostname()
+        # Copy logfiles from non-empty client directories
+        command = "clush -w {} -B -S \"{}\"".format(
+            NodeSet.fromlist(hostlist),
+            "if [ ! -z \\\"\\$(ls -A {0})\\\" ]; then "
+            "scp -p -r {0}/ \\\"{1}:'{2}/'\\\" && rm -rf {0}/*; fi".format(
+                remote_dir, this_host, local_dir))
+        status = process.run(command, timeout=300)
+        return status
+
+    def create_ior_cmdline(self, job_params, job_spec, pool):
+        """Create an IOR cmdline to run in slurm batch.
+
+        Args:
+            job_params (str): job params from yaml file
+            job_spec (str): specific ior job to run
+            pool (obj):   TestPool obj
+
+        Returns:
+            cmd: cmdline string
 
-    def build_ior_script(self, job):
         """
-        Builds an IOR command string which is then added to slurm script
+        iteration = self.test_iteration
+        ior_params = "/run/" + job_spec + "/"
+        if self.test_iteration < 0:
+            iteration = 1000000
+        ior_cmd = IorCommand()
+        ior_cmd.set_params(self, ior_params)
+        ior_cmd.block_size.value = self.params.get(
+            "blocksize", job_params + "*")
+        ior_cmd.repetitions.value = iteration
+        ior_cmd.max_duration.value = self.params.get("time", job_params + '*')
+        ior_cmd.segment_count.value = 1
+        ior_cmd.set_daos_params(self.server_group, pool.pool)
+        # export the user environment to test node
+        exports = ["ALL"]
+        if ior_cmd.api.value == "MPIIO":
+            env = {
+                "CRT_ATTACH_INFO_PATH": os.path.join(
+                    self.basepath, "install/tmp"),
+                "DAOS_POOL": str(ior_cmd.daos_pool.value),
+                "MPI_LIB": "\"\"",
+                "DAOS_SVCL": str(ior_cmd.daos_svcl.value),
+                "DAOS_SINGLETON_CLI": 1,
+                "FI_PSM2_DISCONNECT": 1,
+            }
+            exports.extend(
+                ["{}={}".format(key, val) for key, val in env.items()])
+        command = "srun -l --mpi=pmi2 --export={} {}".format(
+            ",".join(exports), ior_cmd)
+        self.log.debug("<<IOR cmdline >>: %s", command)
+        return command
+
+    def create_dmg_cmdline(self, job_params, job_spec):
```

Comment:
This seems to me some utility function. May be having it in some utility function eg: general_utils might make more sense.